share/eds/store.go

```diff
@@ -71,23 +80,50 @@ func NewStore(basepath string, ds datastore.Batching) (*Store, error) {
 
 	return &Store{
 		basepath: basepath,
-		dgstr:    dagStore,
-		topIdx:   invertedRepo,
-		carIdx:   fsRepo,
-		mounts:   r,
+		// initialize empty gc result to avoid panic on access
+		lastGCResult: &dagstore.GCResult{
+			Shards: make(map[shard.Key]error),
+		},
+		dgstr:  dagStore,
+		topIdx: invertedRepo,
+		carIdx: fsRepo,
+		mounts: r,
 	}, nil
 }
 
-// Start starts the underlying DAGStore.
 func (s *Store) Start(ctx context.Context) error {
+	runCtx, cancel := context.WithCancel(context.Background())
+	s.ctx = runCtx
+	s.cancel = cancel
+
+	go s.gc(runCtx)
 	return s.dgstr.Start(ctx)
 }
 
-// Stop stops the underlying DAGStore.
 func (s *Store) Stop() error {
+	s.cancel()
 	return s.dgstr.Close()
 }
```

Comment:
Please wait for GC to exit before returning

Reply:
Why do you think we need this in this case?

Reply:
Yeah, it's fine because it also survives restarts